kafka-commons/src/test/java/uk/gov/homeoffice/digital/sas/kafka/EmbeddedKafkaIntegrationTest.java

```diff
@@ -43,91 +51,120 @@ class EmbeddedKafkaIntegrationTest {
-    String expectedUpdateMessagePayload = generateExpectedPayload(version, profile, KafkaAction.CREATE);
-    assertThat(kafkaConsumer.getPayload()).isEqualTo(expectedUpdateMessagePayload);
+    assertThat(kafkaConsumerServiceImpl.getKafkaEventMessage()).isNotNull();
+
+    KafkaEventMessage expectedKafkaEventMessage = generateExpectedKafkaEventMessage(version,
+        profile,
+        KafkaAction.CREATE);
+
+    isMessageDeserialized(expectedKafkaEventMessage);
+
   }
 
   @Test
+  @Order(3)
   void shouldSendUpdateMessageToTopicWhenProfileIsUpdated() throws Exception {
-    kafkaConsumer.setExpectedNumberOfMessages(2);
     // GIVEN
     profileRepository.saveAndFlush(profile);
     profile.setName(UPDATED_PROFILE_NAME);
     profile = profileRepository.saveAndFlush(profile);
-
     // WHEN
-    boolean messageConsumed = kafkaConsumer.getLatch().await(CONSUMER_TIMEOUT, TimeUnit.SECONDS);
-
+    kafkaConsumerServiceImpl.setLatch(new CountDownLatch(2));
+    kafkaConsumerServiceImpl.getLatch().await(CONSUMER_TIMEOUT, TimeUnit.SECONDS);
     // THEN
-    assertThat(messageConsumed).isTrue();
-    String expectedUpdateMessagePayload = generateExpectedPayload(version, profile, KafkaAction.UPDATE);
-    assertThat(kafkaConsumer.getPayload()).isEqualTo(expectedUpdateMessagePayload);
+    assertThat(kafkaConsumerServiceImpl.getKafkaEventMessage()).isNotNull();
+
+    KafkaEventMessage expectedKafkaEventMessage = generateExpectedKafkaEventMessage(version,
+        profile,
+        KafkaAction.UPDATE);
+
+    isMessageDeserialized(expectedKafkaEventMessage);
   }
 
   @Test
+  @Order(4)
   void shouldSendDeleteMessageToTopicWhenProfileIsDeleted() throws Exception {
-    kafkaConsumer.setExpectedNumberOfMessages(2);
     // GIVEN
     profileRepository.save(profile);
     profileRepository.delete(profile);
-
     // WHEN
-    boolean messageConsumed = kafkaConsumer.getLatch().await(CONSUMER_TIMEOUT, TimeUnit.SECONDS);
-
+    kafkaConsumerServiceImpl.setLatch(new CountDownLatch(2));
+    kafkaConsumerServiceImpl.getLatch().await(CONSUMER_TIMEOUT, TimeUnit.SECONDS);
     // THEN
-    assertThat(messageConsumed).isTrue();
-    String expectedDeleteMessagePayload = generateExpectedPayload(version, profile, KafkaAction.DELETE);
-    assertThat(kafkaConsumer.getPayload()).isEqualTo(expectedDeleteMessagePayload);
+    assertThat(kafkaConsumerServiceImpl.getKafkaEventMessage()).isNotNull();
+
+    KafkaEventMessage expectedKafkaEventMessage = generateExpectedKafkaEventMessage(version,
+        profile,
+        KafkaAction.DELETE);
+
+    isMessageDeserialized(expectedKafkaEventMessage);
+  }
+
+  private KafkaEventMessage generateExpectedKafkaEventMessage(String version, Profile resource,
+                                                              KafkaAction action) {
+    return new KafkaEventMessage<>(version, resource, action);
+  }
+
+  private void isMessageDeserialized(KafkaEventMessage expectedKafkaEventMessage) {
```

Comment:
This method is not returning a boolean. It should be renamed accordingly. e.g `assertMessageIsDeserialisedAsExpected`